src/blueprints/tv2/action-factories/tv2-transition-effect-action-factory.ts

```diff
@@ -147,9 +147,9 @@ export class Tv2TransitionEffectActionFactory extends ActionFactory {
       }
     }
 
-    const updateTransitionMutateAction: MutateActionWithPieceMethods = {
+    const updateTransitionMutateAction: MutateActionWithOnAirAndNextPiecesMethods = {
       type: MutateActionType.PIECE,
-      updateActionWithPiece: (action: Action, piece: Piece) => this.updateTimelineObjectsWithTransitionEffect(action as Tv2TransitionEffectAction, piece),
+      updateActionWithPiece: (action: Action, onAirPiece: Piece | undefined, nextPiece: Piece | undefined) => this.updateTimelineObjectsWithTransitionEffect(action as Tv2TransitionEffectAction, onAirPiece, nextPiece),
```

Comment:
```suggestion
      updateActionWithPiece: (action: Action, onAirPiece?: Piece, nextPiece?: Piece) => this.updateTimelineObjectsWithTransitionEffect(action as Tv2TransitionEffectAction, onAirPiece, nextPiece),
```

how do you feel about using optional instead of using undefined? :)

Reply:
That is something we like to use.
Updated.